packages/components/src/date-time/test/date.js

```diff
@@ -1,97 +1,118 @@
+
+		const todayDescription = moment().format( 'dddd, MMM D, YYYY' );
+		expect( screen.getByLabelText( todayDescription ).classList ).toContain(
+			'CalendarDay__selected'
+		);
 	} );
 
-	it( 'should pass down a moment object for now when currentDate is undefined', () => {
-		const wrapper = shallow( <DatePicker /> );
-		const date = wrapper.children().props().date;
-		expect( moment.isMoment( date ) ).toBe( true );
+	it( 'should call onChange when a day is selected', async () => {
+		const user = userEvent.setup( { delay: null } );
+
+		const onChange = jest.fn();
+
+		render(
+			<DatePicker
+				currentDate="2022-05-02T11:00:00"
+				onChange={ onChange }
+			/>
+		);
+
+		await user.click( screen.getByLabelText( 'Friday, May 20, 2022' ) );
+
+		expect( onChange ).toHaveBeenCalledWith( '2022-05-20T11:00:00' );
 	} );
 
-	describe( 'onChangeMoment', () => {
-		it( 'should call onChange with a formated date of the input', () => {
-			const onChangeSpy = jest.fn();
-			const currentDate = '1986-10-18T11:00:00';
-			const wrapper = shallow(
-				<DatePicker
-					currentDate={ currentDate }
-					onChange={ onChangeSpy }
-				/>
-			);
-			const newDate = moment();
-
-			wrapper.childAt( 0 ).props().onDateChange( newDate );
-
-			expect( onChangeSpy ).toHaveBeenCalledWith(
-				newDate.format( TIMEZONELESS_FORMAT )
-			);
-		} );
-
-		it( 'should call onChange with hours, minutes, seconds of the current time when currentDate is undefined', () => {
-			let onChangeSpyArgument;
-			const onChangeSpy = ( arg ) => ( onChangeSpyArgument = arg );
-			const wrapper = shallow( <DatePicker onChange={ onChangeSpy } /> );
-			const newDate = moment( '1986-10-18T11:00:00' );
-			const current = moment();
-			const newDateWithCurrentTime = newDate.clone().set( {
-				hours: current.hours(),
-				minutes: current.minutes(),
-				seconds: current.seconds(),
-			} );
-			wrapper.childAt( 0 ).props().onDateChange( newDate );
-
-			expect(
-				moment( onChangeSpyArgument ).isSame(
-					newDateWithCurrentTime,
-					'minute'
-				)
-			).toBe( true );
-		} );
-
-		it( 'should call onChange with hours, minutes, seconds of the current time when currentDate is null', () => {
-			let onChangeSpyArgument;
-			const onChangeSpy = ( arg ) => ( onChangeSpyArgument = arg );
-			const wrapper = shallow(
-				<DatePicker currentDate={ null } onChange={ onChangeSpy } />
-			);
-			const newDate = moment( '1986-10-18T11:00:00' );
-			const current = moment();
-			const newDateWithCurrentTime = newDate.clone().set( {
-				hours: current.hours(),
-				minutes: current.minutes(),
-				seconds: current.seconds(),
-			} );
-			wrapper.childAt( 0 ).props().onDateChange( newDate );
-
-			expect(
-				moment( onChangeSpyArgument ).isSame(
-					newDateWithCurrentTime,
-					'minute'
-				)
-			).toBe( true );
-		} );
+	it( 'should call onMonthPreviewed and onChange when a day in a different month is selected', async () => {
+		const user = userEvent.setup( { delay: null } );
```

Comment:
Would you mind making a few changes to these tests in order to align to how we use `@testing-library/user-event` in other components?

You'd need to rebase on top of `trunk` to include [this version bump to `user-event`](https://github.com/WordPress/gutenberg/pull/40761) (don't forget to reinstall `node_modules`), and then switch to use jest fake timers + the `advanceTimers` option from `userEvent`

<details>
<summary>Example</summary>

```diff
diff --git a/packages/components/src/date-time/test/date.js b/packages/components/src/date-time/test/date.js
index f204dff80f..4cf66697f4 100644
--- a/packages/components/src/date-time/test/date.js
+++ b/packages/components/src/date-time/test/date.js
@@ -11,7 +11,20 @@ import 'react-dates/initialize';
  */
 import DatePicker from '../date';
 
+const user = userEvent.setup( {
+	advanceTimers: jest.advanceTimersByTime,
+} );
+
 describe( 'DatePicker', () => {
+	beforeEach( () => {
+		jest.useFakeTimers();
+	} );
+
+	afterEach( () => {
+		jest.runOnlyPendingTimers();
+		jest.useRealTimers();
+	} );
+
 	it( 'should highlight the current date', () => {
 		render( <DatePicker currentDate="2022-05-02T11:00:00" /> );
 
@@ -34,8 +47,6 @@ describe( 'DatePicker', () => {
 	} );
 
 	it( 'should call onChange when a day is selected', async () => {
-		const user = userEvent.setup( { delay: null } );
-
 		const onChange = jest.fn();
 
 		render(
@@ -51,8 +62,6 @@ describe( 'DatePicker', () => {
 	} );
 
 	it( 'should call onMonthPreviewed and onChange when a day in a different month is selected', async () => {
-		const user = userEvent.setup( { delay: null } );
-
 		const onMonthPreviewed = jest.fn();
 		const onChange = jest.fn();
 
@@ -99,8 +108,6 @@ describe( 'DatePicker', () => {
 	} );
 
 	it( 'should not allow invalid date to be selected', async () => {
-		const user = userEvent.setup( { delay: null } );
-
 		const onChange = jest.fn();
 
 		render(
```
</details>

I just merged a PR with similar changes today (#40790).

Also happy if you want to carry these changes in a follow-up !

Reply:
Opened a follow-up for this: https://github.com/WordPress/gutenberg/pull/40800

Unfortunately the change causes a weird error in `react-dates`. Let's discuss in the other PR.